### DISCUSSION TITLE
Update variable value

### DISCUSSION THREAD

**Discussion by cojoclaudiu** [vanilla-extract-css/vanilla-extract#1046] [Q&A] [answered]
Hello,

I have a question regarding with the CSS variables in Vanilla Extract.

I've get this
```ts
export const theme = createGlobalTheme(':root', {
	colors: {
		currentColor: 'red',
...
}
```

I would like to update the `theme.colors.currentColor` inside the `useEffect` with a color coming from the API call.

How I can do that?

**Comment by askoufis** [comment, marked as answer]
Discussed this on [discord](https://discord.com/channels/885877446098964512/1087681018418429993/1087681249646223410), the option that made the most sense in this case was to use [`setElementVars`](https://vanilla-extract.style/documentation/packages/dynamic/#setelementvars) on the document element, for example:

```tsx
setElementVars(document.documentElement, {
  [theme.colors.currentColor]: 'blue'
});
```

**Reply by cojoclaudiu** [reply]
@askoufis thank you again for help, in my case that is the solution
